### ISSUE TITLE
[Android] Opening Brave Stats flyout causes Brave to no longer honor rotation lock

### ISSUE THREAD

**Issue by intrnl** [brave/brave-browser#11059]
## Description 
Opening Brave Stats flyout on NTP causes Brave to no longer honor the rotation lock

## Steps to reproduce 

   1. Make sure auto-rotate is disabled (rotation lock)
   2. Open a new tab
   3. Open the Brave Stats flyout and close it
   4. Turn the device sideways

## Actual result 
The rotation lock is ignored and the screen rotates to landscape

## Expected result
The rotation lock is honored and it doesn't rotate the screen

## Issue reproduces how often 
Easily reproduced

## Issue happens on 
   - Current Play Store version? 1.14.10
   - Beta build? Yes, Nightly

## Device details
   - Install type (ARM, x86): arm64
   - Device (Phone, Tablet, Phablet): Phone
   - Android version: 10
   
## Brave version
1.14.10 Nightly

**Comment by srirambv** [comment]
This seems expected. When device is in landscape mode there is no way to populate the stats flyout. 

cc: @jamesmudgett

**Comment by intrnl** [comment]
Brave Stats flyout forcing portrait might be expected, what's unexpected for me is that turning the device sideways after closing the flyout shouldn't have put it to landscape since I have the rotation lock.

**Comment by intrnl** [comment]
I have to close the app and reopen it again just so my screen stays locked on portrait.

**Comment by bbolgradov** [comment]
I can confirm this issue. Really annoying.

**Comment by srirambv** [comment]
Ok, I see one potential annoying factor on a tablet as well but not because of portrait. If Auto-rotate is enabled and the device is rotated 180deg in portrait mode, clicking on stats rotates it back to the original orientation

**Comment by acadet** [comment]
Any update on this issue? It's happening to me as well

```
Brave 1.13.87, Chromium 85.0.4183.102
Android 9; ONEPLUS A3000 Build/PKQ1.181203.001
```

**Comment by intrnl** [comment]
This still happens on 1.16.17

**Comment by intrnl** [comment]
Should be fixed by brave/brave-core#6650
